### PR TITLE
docs: add storybook support

### LINKED ISSUE #521: docs: add Storybook (or some other component workbench)
- Tried it once: #511 
- Blocked by https://github.com/storybookjs/storybook/issues/22065

### PR DESCRIPTION
**Pull request by hollandjg**
Add Storybook (https://storybook.js.org) support and "stories" for all of the simple components.

The storybook and all of its config is in the packages/storybook directory to isolate its dependencies and to avoid any unwanted regressions in the behavior of the other packages (example-site, defaults etc.).

## Broken
- Images
- Things which rely on staticQuery:
  - components/SiteMetadata 
  - components/ProjectDetail (due to using the `useSiteMetadata` hook to load the project interest link)
  - all Layouts (due to their using SiteMetadata)

## Related issues etc.
- resolves #521

### CODE REVIEW

File: packages/storybook/.storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+// Navigating through a gatsby app using gatsby-link or any other gatsby component will use the `___navigate` method.
+// In Storybook, it makes more sense to log an action than doing an actual navigate. Check out the actions addon docs for more info: https://storybook.js.org/docs/react/essentials/actions
+
+window.___navigate = (pathname) => {
+  action("NavigateTo:")(pathname)
+}
+
+import "@thepolicylab-projectportals/gatsby-theme-project-portal/src/styles/global.css"
+/** @type { import('@storybook/react').Preview } */
```
Comment (hetd54) [inline packages/storybook/.storybook/preview.js:23]: do you need this still?

File: packages/storybook/.storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      matchers: {
+        color: /(background|color)$/i,
+        date: /(Date|opportunityCloses|created|lastModified)$/,
+      },
+    },
+  },
+
+  decorators: [
+    // Adds theme switching support.
+    // NOTE: requires setting "darkMode" to "class" in your tailwind config
```
Comment (hetd54) [inline packages/storybook/.storybook/preview.js:37]: should we update our tailwind config in this pr?

**Review by hetd54 — APPROVED** [review]
I have a couple questions, but nothing that should hold you back!

File: packages/storybook/.storybook/preview.js (hunk 1 above)
Comment (hollandjg) [reply packages/storybook/.storybook/preview.js:23]: Yes, this is a weird JS type-hint. I couldn't manage to get the thing working as TS (which would have more understandable type hints), but we should probably do that. Something with the Babel configuration made it tricky – I kept getting errors.

File: packages/storybook/.storybook/preview.js (hunk 2 above)
Comment (hollandjg) [reply packages/storybook/.storybook/preview.js:37]: Ooh, I'd not even seen that. It didn't work at the first attempt, but we might be able to get it working. I think that will be a new issue.

**Pull request merged**